src/Internal/QueryM/Sign.purs

```diff
@@ -44,57 +35,10 @@ signTransaction tx = do
     Nami nami -> liftAff $ callCip30Wallet nami \nw -> flip nw.signTx tx
     Gero gero -> liftAff $ callCip30Wallet gero \nw -> flip nw.signTx tx
     Flint flint -> liftAff $ callCip30Wallet flint \nw -> flip nw.signTx tx
-    Eternl eternl -> do
-      let
-        txInputs :: Array TransactionInput
-        txInputs = fromFoldable $ tx ^. _body <<< _inputs
-      walletWaitForInputs txInputs
-      liftAff $ callCip30Wallet eternl \nw -> flip nw.signTx tx
+    Eternl eternl -> liftAff $ callCip30Wallet eternl \nw -> flip nw.signTx tx
     Lode lode -> liftAff $ callCip30Wallet lode \nw -> flip nw.signTx tx
     NuFi nufi -> liftAff $ callCip30Wallet nufi \w -> flip w.signTx tx
     KeyWallet kw -> liftAff do
       witnessSet <- (unwrap kw).signTx tx
       pure $ Just (tx # _witnessSet <>~ witnessSet)
 
--- | Waits till all provided inputs of a given transaction appear in the UTxO
--- | set provided by the wallet.
--- | This is a hacky solution to the problem of Eternl not seeing UTxOs that
--- | hasn't been fully confirmed at the moment of a `sign()` call.
--- | Since it can't detect UTxO origin, it can't decide which of the private
--- | keys to use for signing. As a result, we get `MissingVKeyWitnesses`.
-walletWaitForInputs :: Array TransactionInput -> QueryM Unit
-walletWaitForInputs txInputs = do
```

Comment:
Could you please remind of why we decided that this was no longer necessary?

Reply:
What if we waited for all the outputs of a Tx that belong to self show up in `getWalletUtxos` at the `awaitTxConfirmed` stage?

Reply:
No guarantee that the utxos are not spent in another Contract or app.